spec/controllers/admin/versions_controller_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'spec_helper'
+
+describe Admin::VersionsController do
+
+  let(:conference) { create(:conference, short_title: 'exampletitle', description: 'Example Description') }
+  let(:admin) { create(:admin) }
+
+  with_versioning do
+    describe 'GET #revert' do
+      before :each do
+        sign_in admin
+      end
+
+      context 'when no attribute is set' do
+        it 'reverts all changes for update events' do
+          conference.update_attributes(short_title: 'testtitle', description: 'Some random text')
+          get :revert, id: PaperTrail::Version.last.id
+          conference.reload
+          expect(conference.short_title).to eq 'exampletitle'
+          expect(conference.description).to eq 'Example Description'
+        end
+
+        it 'shows correct flash on trying to revert create event of a deleted object' do
+          creation_version_id = conference.program.event_types.first.versions.first.id
+          conference.program.event_types.first.destroy
+          get :revert, id: creation_version_id
+          expect(flash[:error]).to match('The item is already in the state that you are trying to revert it back to')
+        end
+
+        it 'reverting deletion of object  creates it again' do
+          conference.program.event_types.first.destroy
+          get :revert, id: PaperTrail::Version.last.id
+          conference.reload
+          expect(PaperTrail::Version.last.event).to eq 'create'
+          expect(conference.program.event_types.count).to eq 2
+        end
```

Comment:
I would like to know the count of event_types *prior* to the action we are testing, so that this tests makes more sense when reading it.